inference-engine/tests/functional/inference_engine/transformations/remove_concat_zero_dim_input_test.cpp

```diff
@@ -0,0 +1,196 @@
+    std::shared_ptr<ov::Function> f(nullptr), f_ref(nullptr);
+    auto input1 = std::make_shared<ov::opset8::Parameter>(ov::element::f32, ov::PartialShape{1, ov::Dimension::dynamic(), 3});
+    auto input3 = std::make_shared<ov::opset8::Parameter>(ov::element::f32, ov::PartialShape{1, 2, 3});
+    auto abs = std::make_shared<ov::opset8::Abs>(input1);
+    int64_t axis = 1;
+    {
+        auto in_mul = std::make_shared<ov::opset8::Parameter>(ov::element::f32, ov::PartialShape{1, 0, 3});
+        auto mul = std::make_shared<ov::opset8::Multiply>(in_mul, abs);
+        auto concat = std::make_shared<ov::opset8::Concat>(ov::OutputVector{mul, input3}, axis);
+
+        f = std::make_shared<ov::Function>(ov::NodeVector{concat}, ov::ParameterVector{input1, input3, in_mul});
+
+        ov::pass::Manager manager;
+        manager.register_pass<ngraph::pass::InitNodeInfo>();
+        manager.register_pass<ov::pass::RemoveConcatZeroDimInput>();
+        manager.run_passes(f);
+        ASSERT_NO_THROW(check_rt_info(f));
+    }
+
+    {
+        auto concat = std::make_shared<ov::opset8::Concat>(ov::OutputVector{input3}, axis);
+        f_ref = std::make_shared<ov::Function>(ov::NodeVector{concat}, ov::ParameterVector{input3});
+    }
+
+    const auto fc = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const auto res = fc.compare(f, f_ref);
+    ASSERT_TRUE(res.valid) << res.message;
+}
+
+TEST(TransformationTests, RemoveConcatZeroDimInputPartiallyKnowShape) {
+    std::shared_ptr<ov::Function> f(nullptr), f_ref(nullptr);
+    auto input1 = std::make_shared<ov::opset8::Parameter>(ov::element::f32, ov::PartialShape::dynamic());
+    auto input3 = std::make_shared<ov::opset8::Parameter>(ov::element::f32, ov::PartialShape::dynamic());
+    int64_t axis = 0;
+    {
+        auto input2 = std::make_shared<ov::opset8::Parameter>(ov::element::f32,
+            ov::PartialShape{0, ov::Dimension::dynamic(), ov::Dimension::dynamic()});
+        auto concat = std::make_shared<ov::opset8::Concat>(ov::OutputVector{input1, input2, input3}, axis);
+
+        f = std::make_shared<ov::Function>(ov::NodeVector{concat}, ov::ParameterVector{input1, input2, input3});
+
+        ov::pass::Manager manager;
+        manager.register_pass<ngraph::pass::InitNodeInfo>();
+        manager.register_pass<ov::pass::RemoveConcatZeroDimInput>();
+        manager.run_passes(f);
+        ASSERT_NO_THROW(check_rt_info(f));
+        f->validate_nodes_and_infer_types();
+    }
+
+    {
+        auto concat = std::make_shared<ov::opset8::Concat>(ov::OutputVector{input1, input3}, axis);
+        f_ref = std::make_shared<ov::Function>(ov::NodeVector{concat}, ov::ParameterVector{input1, input3});
+    }
+
+    const auto fc = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const auto res = fc.compare(f, f_ref);
+    ASSERT_TRUE(res.valid) << res.message;
+}
+
+TEST(TransformationTests, RemoveConcatZeroDimInputDynamicRank) {
```

Comment:
We use `TEST_F(TransformationsTestsF,` for such type of tests.

Reply:
done, but I had problems with tests:
- RemoveConcatZeroDimInputPartiallyKnowShape
- RemoveConcatZeroDimTwoInputs

For these cases output shapes are other than expected (validate and infer type is not called because we are returning always now "false" in transformation). Manually registering `Validate` pass also doesn't help (`function_changed` in `manager.cpp` is `false`). I can't manually call `f->validate_nodes_and_infer_types()` in `TEST_F(TransformationTestsF` approach before results comparison.

Reply:
as discussed offline, added returning status to `RemoveConcatZeroDimInput`